ee/tabby-webserver/src/mock/mod.rs

```diff
@@ -0,0 +1,180 @@
+pub mod helper;
+
+use async_openai::{
+    error::OpenAIError,
+    types::{
+        ChatChoice, ChatChoiceStream, ChatCompletionResponseMessage, ChatCompletionResponseStream,
+        ChatCompletionStreamResponseDelta, CompletionUsage, CreateChatCompletionRequest,
+        CreateChatCompletionResponse, CreateChatCompletionStreamResponse, FinishReason, Role,
+    },
+};
+use axum::async_trait;
+use tabby_common::api::{
+    code::{CodeSearch, CodeSearchError, CodeSearchParams, CodeSearchQuery, CodeSearchResponse},
+    doc::{DocSearch, DocSearchDocument, DocSearchError, DocSearchHit, DocSearchResponse},
+};
+use tabby_inference::ChatCompletionStream;
+use tabby_schema::{
+    context::{ContextInfo, ContextService},
+    policy::AccessPolicy,
+    Result,
+};
+
+pub struct MockChatCompletionStream;
```

Comment:
```suggestion
pub struct FakeChatCompletionStream;
```

Usually we call an implementation `Mock` if it's behavior can be defined in testing time (e.g mockall: https://docs.rs/mockall/latest/mockall/).

For implementation with fixed output (for testing), usually we call it `FakeXXXX`